boot/zephyr/main.c

```diff
@@ -52,7 +52,7 @@ const struct boot_uart_funcs boot_funcs = {
 #endif
 
 #if defined(CONFIG_LOG) && !defined(CONFIG_LOG_IMMEDIATE) && \
-    !defined(CONFIG_LOG_MINIMAL)
+    !defined(CONFIG_LOG_MODE_MINIMAL)
```

Comment:
Do you know the specifics of when this was changed in Zephyr, and what we need to do to support multiple Zephyr versions. One possibility is to conditionalize on either definition. e.g. `&& !(defined(CONFIG_LOG_MINIMAL) || defined(CONFIG_LOG_MODE_MINIMAL))`.

Reply:
In light of this, we could also conditionalize this once, and define a local definition that would keep all of this from needing to be expanded in every place we care about it (which, admittedly is only two at this point).

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/31404 seem to introduce the change. To be honest I'm not sure if multiple versions have to be supported, Zephyr maintainers or authors of the changes should be able to tell.

Reply:
There is also the `prj.conf` file change, which cannot easily support both.

Reply:
>In light of this, we could also conditionalize this once, and define a local definition that would keep all of this from needing to be expanded in every place we care about it (which, admittedly is only two at this point).

more is not expected

I encourage to keep backward compatibility in the code.

CONFIG_LOG_MINIMAL is still emitted by zephyr, change here is not required or better check can be just extended by new define check.